Realtime/src/commonMain/kotlin/io/github/jan/supabase/realtime/PostgrestExtensions.kt

```diff
@@ -0,0 +1,86 @@
+package io.github.jan.supabase.realtime
+
+import io.github.jan.supabase.postgrest.query.PostgrestQueryBuilder
+import io.github.jan.supabase.postgrest.query.filter.FilterOperation
+import io.github.jan.supabase.postgrest.query.filter.PostgrestFilterBuilder
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.emitAll
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.onCompletion
+import kotlin.reflect.KProperty1
+
+/**
+ * Executes vertical filtering with select on [PostgrestQueryBuilder.table] and [PostgrestQueryBuilder.schema] and returns a [Flow] of a single value matching the [filter].
+ * This function listens for changes in the table and emits the new value whenever a change occurs.
+ * @param primaryKey the primary key of the [Data] type
+ * @param filter the filter to apply to the select query
+ */
+inline fun <reified Data : Any> PostgrestQueryBuilder.selectSingleValueAsFlow(
+    primaryKey: PrimaryKey<Data>,
+    crossinline filter: PostgrestFilterBuilder.() -> Unit
+): Flow<Data> {
+    val realtime = postgrest.supabaseClient.realtime as RealtimeImpl
+    val id = realtime.nextIncrementId()
+    val channel = realtime.channel("$schema:$table:$id")
```

Comment:
I'd say this could be part of the API as optional if someone wants to listen to some shared channel, if not provided, use what you have here.